django_celery_beat/schedulers.py

```diff
@@ -280,6 +285,8 @@ def schedule(self):
         if update:
             self.sync()
             self._schedule = self.all_as_schedule()
+            # the schedule changed, invalidate the heap in Scheduler.tick
+            self._heap = None
```

Comment:
I'm aware that this is private to the parent class Scheduler and the tick function. Maybe the tick function can have a method to call which indicates if the schedule has been changed since it last ran, if the schedule has changed, then the heap needs to be invalidated because it now has un-sycned task entries